subsys/bluetooth/host/att.c

```diff
@@ -3088,9 +3105,42 @@ int bt_eatt_connect(struct bt_conn *conn, uint8_t num_channels)
 		return -ENOMEM;
 	}
 
-	return bt_l2cap_ecred_chan_connect(conn, chan, BT_EATT_PSM);
+	while (created < i) {
+		/* bt_l2cap_ecred_chan_connect() uses the first L2CAP_ECRED_CHAN_MAX_PER_REQ
+		 * elements of the array or until a null-terminator is reached.
+		 */
+		err = bt_l2cap_ecred_chan_connect(conn, &chan[created], BT_EATT_PSM);
+		if (err < 0) {
+			return err;
+		}
+
+		created += L2CAP_ECRED_CHAN_MAX_PER_REQ;
+	}
+
+	return 0;
 }
 
+#if defined(CONFIG_BT_EATT) && defined(CONFIG_BT_EATT_AUTO_CONNECT)
```

Comment:
Does `CONFIG_BT_EATT_AUTO_CONNECT` not depend on `CONFIG_BT_EATT`? If so, this can just be guarded by `CONFIG_BT_EATT_AUTO_CONNECT`, and if not, it probably should be

Reply:
It is hidden behind a `if BT_EATT` in the Kconfig. Should a `depends on BT_EATT` be added to this and the other EATT Kconfig options?

It is not possible to select `CONFIG_BT_EATT_AUTO_CONNECT` without `CONFIG_BT_EATT`, so the `#ifdef` can be simplified.

Reply:
> Should a depends on BT_EATT be added to this and the other EATT Kconfig options?

Not if it is already hidden by a `if BT_EATT` :)